aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -2646,17 +2646,16 @@ Tensor mexp_impl(
         a_large_norm,
         large_norm_subset,
         thetas[total_n_degs - 1]
-      );
+      ).to(a.dtype());
```

Comment:
Is this really necessary?

Reply:
"Index put requires the source and destination dtypes match", this should be necessary.